### DISCUSSION TITLE
Packet timing problem

### DISCUSSION THREAD

**Discussion by junhyeok-yoon** [ANLAB-KAIST/KENSv3#168] [Project Q&A 2. Transport Layer (TCP)] [answered]
First, I'm really grateful for TA's detailed and kind answers. I think I seem to be helping me do well on my assignment.

I make my code has same packet log(order, flags, seq#, ack#, port, ip...) with the solution.
Also, every packekt timing is identical to the solution.
EXCEPT, close _connect_* test cases. First figure is the solution and second is my answer.
1. Is this problematic? And I want to know why this happen.
(solution)

![image](https://github.com/ANLAB-KAIST/KENSv3/assets/114990411/ad67c551-f794-470a-a99c-3e0e9738e5e1)

(my answer)

![image](https://github.com/ANLAB-KAIST/KENSv3/assets/114990411/e8b893fb-85a3-4c8f-83f1-c125e5667bf8)

2. Should I check whether the all filedecriptors are removed?(even though pass open/close test)

3. I pass all 4 test code, and have same order of packet and its info.(Except for timing problem in 1)
Am I done for PA2?

**Comment by Pusnow** [comment, marked as answer]
Would you give me your testing log? The log describes why you are failing.

> 2. Should I check whether the all filedecriptors are removed?(even though pass open/close test)

If you pass the all test cases, you will get full credit. However, please try to implement all features of TCP stack as much as possible for your education.

> 3. I pass all 4 test code, and have same order of packet and its info.(Except for timing problem in 1)
> Am I done for PA2?

If you pass all test cases in `test-kens-open`, `test-kens-bind`, `test-kens-handshake`, and `test-kens-close`, you have all done for PA2.

**Reply by junhyeok-yoon** [reply]
사진이 업로드 되지 않아 수정했습니다!

**Reply by Pusnow** [reply]
I meant the log messages printed in the terminal.

**Reply by junhyeok-yoon** [reply]
Belows are the terminal after executing tests. I pass all of them.
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/114990411/a45f8282-0868-46e0-9d72-a392df4106e8)

![image](https://github.com/ANLAB-KAIST/KENSv3/assets/114990411/9d89ed7a-f8dc-41f5-a762-ea2edda678c8)

![image](https://github.com/ANLAB-KAIST/KENSv3/assets/114990411/fd75178a-b3fe-41f5-9da1-d2b718071e39)

**Reply by Pusnow** [reply]
If you pass all test cases, you will get full credit. Timing is not important for PAs.
